### ISSUE TITLE
can't get past login page

### ISSUE THREAD

**Issue by fadelm0** [bfabiszewski/ulogger-server#5]
The user isn't getting authenticated through the login page. I'm getting a wrong username or password, even though I put them everywhere I'm supposed to.

**Comment by amadeous** [comment]
Hey !
How did you performed the install instructions ?

**Comment by fadelm0** [comment]
Exactly as written, except used another name and password.

So, lets say i used "abc" for username and "12345" for password.
I first added this user using phpmyAdmin, with all permissions set.

In config.php, used the following credentials:
```
$dbhost = "localhost";
$dbuser = "abc";
$dbpass = "12345";
$dbname = "ulogger";`
```

In ulogger.sql I added these lines:
```
INSERT INTO `users` (`id`, `login`, `password`) VALUES
(1, 'abc', '12345');
```

Although, I have to mention that for some reason after importing the database, the entry did not get added to the table automatically (should i create another issue for this?). So, I had to manually enter the mysql into phpmyAdmin.

I then go to localhost/ulogger-server/, get the login page, enter **"abc"** as username and **"12345"** as password, and get the **"wrong username or password"** error.

**Comment by fadelm0** [comment]
okay, I followed it exactly with the exact admin and pass, and it worked this time. I just didn't understand why the password was so long, and how was I able to login using admin/admin?

**Comment by amadeous** [comment]
The configuration for accessing the database seems to be right if you have an authentification problem.

Adding this insert line in the ulogger.sql file wasn't a bad idea. The only problem is that the password are first hashed and then stored (as you can see it in the [adduser.php](https://github.com/bfabiszewski/ulogger-server/blob/master/adduser.php) file at line 45). The password you entered is not hashed. When you try to connect with 12345, mysql compares 12345 with $2y$10$ycdU42aG.5A.7.lXY9RKlual16iT0TKMbks7mzUakb1W9yfUFtw9K which is the hash of 12345.

In short : you have to change the abc password in the database for $2y$10$ycdU42aG.5A.7.lXY9RKlual16iT0TKMbks7mzUakb1W9yfUFtw9K instead of 12345. Then when you try to login with abcd/12345 it should work.

**Comment by bfabiszewski** [comment]
Passwords are not stored on the server. The long string you see is password hash.

**Comment by fadelm0** [comment]
@amadeous I see :) 
This is my first experience with web apps and login pages. Thanks for the explanation!

**Comment by amadeous** [comment]
@fadelm0 With pleasure ! ;)
